test/e2e/networkingnew.go

```diff
@@ -0,0 +1,358 @@
+		//copy to local to avoid range overwriting
+		timeoutSeconds := svcSoak.timeoutSeconds
+		serviceNum := svcSoak.service
+		It(fmt.Sprintf("should function for intrapod communication between all hosts in %v parallel services", serviceNum),
+			func() {
+				Logf("running service test with timeout = %v for %v", timeout, serviceNum)
+				runNetTest(timeoutSeconds, f, makePorts(serviceNum), nettestVersion)
+			})
+	}
+})
+
+//pollPeerStatus will either fail, pass, or continue polling.
+//When completed, it will write the service name to the channel provided, thus
+//facilitating parallel service testing.
+func pollPeerStatus(serviceDoneChannel chan string, f *Framework, svc *api.Service, pollTimeoutSeconds time.Duration) {
+
+	Logf("Begin polling " + svc.Name)
+	getDetails := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("read").
+			DoRaw()
+	}
+
+	getStatus := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("status").
+			DoRaw()
+	}
+
+	passed := false
+
+	expectNoError(wait.Poll(2*time.Second, pollTimeoutSeconds, func() (bool, error) {
+		body, err := getStatus()
+		if err != nil {
+			Failf("Failed to list nodes: %v", err)
+		}
+
+		// Finally, we pass/fail the test based on if the container's response body, as to whether or not it was able to find peers.
+		switch {
+		case string(body) == "pass":
+			passed = true
+			return true, nil
+		case string(body) == "running":
+		case string(body) == "fail":
+			if body, err = getDetails(); err != nil {
+				Failf("Failed on attempt. Error reading details: %v", err)
+				return false, err
+			} else {
+				Failf("Failed on attempt. Details:\n%s", string(body))
+				return false, nil
+			}
+		case strings.Contains(string(body), "no endpoints available"):
+			Logf("Attempt: waiting on service/endpoints")
+		default:
+			Logf("Unexpected response:\n%s", body)
+		}
+		return false, nil
+	}))
+
+	if !passed {
+		if body, err := getDetails(); err != nil {
+			Failf("Timed out.  Major error : Couldn't read service details: %v", err)
+		} else {
+			Failf("Timed out. Service details :\n%s", string(body))
+		}
+	}
+	serviceDoneChannel <- svc.Name
+}
+
+//runNetTest Creates a single pod on each host which serves
+//on a unique port in the cluster.  It then binds a service to
+//that port, so that there are "n" nodes to balance traffic to -
+//finally, each node reaches out to ping every other node in
+//the cluster on the given port.
+//The more ports given, the more services will be spun up,
+//i.e. one service per port.
+//To test basic pod networking, send a single port.
+//To soak test the services, we can send a range (i.e. 8000-9000).
+func runNetTest(timeoutSeconds time.Duration, f *Framework, ports []int, nettestVersion string) {
+	defer GinkgoRecover()
```

Comment:
It doesn't look like `runNetTest` is run in a goroutine. I believe this isn't needed here, and instead needs to be moved to  just below line 237 (`go func(nodes *api.NodeList, port int) {`)